packages/@n8n/nodes-langchain/nodes/llms/N8nLlmTracing.ts

```diff
@@ -83,7 +85,9 @@ export class N8nLlmTracing extends BaseCallbackHandler {
 		return encodedListLength.reduce((acc, curr) => acc + curr, 0);
 	}
 
-	async handleLLMEnd(output: LLMResult) {
+	async handleLLMEnd(output: LLMResult, runId: string) {
+		const runDetails = this.runsQueue[runId] ?? { index: Object.keys(this.runsQueue).length - 1 };
```

Comment:
Is it possible that langchain calls `handleLLMEnd` without calling `handleLLMStart` before?
Also, do we need to subtract 1? (if we assume that existing records have sequential indexes starting from 0 – the next index will be equal to `length`)

Reply:
I don't think that's possible,  `handleLLMStary` hook should always be called before `handleLLMEnd` but just in case it somehow happened, I wanted to have at least index fallback. Fair point with not needing -1 in that case